packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -210,32 +217,49 @@ export class ConnectionStateHandler {
     }
 
     public receivedDisconnectEvent(reason: string) {
-        if (this.joinOpTimer.hasTimer) {
-            this.stopJoinOpTimer();
-        }
         this.setConnectionState(ConnectionState.Disconnected, reason);
     }
 
+    private readonly transitionToConnectedState = () => {
+        // Defensive measure, we should always be in Connecting state when this is called.
+        const inConnectingState = this._connectionState === ConnectionState.Connecting;
+        logIfFalse(
+            inConnectingState,
+            this.logger,
+            { eventName: "cannotTransitionToConnectedState", connectionState: ConnectionState[this._connectionState] });
+        if (!inConnectingState) {
+            return;
+        }
+
+        this.setConnectionState(ConnectionState.Connected);
+    };
+
     /**
-     * The "connect" event indicates the connection to the Relay Service is live.
+     * The "connect" event indicates the connection to the ordering service is live.
```

Comment:
We are moving towards "relay service" terminology than "ordering service". So, we should revert this change.